### PR TITLE
chore(license): move third party licenses to separate file

### PR DESCRIPTION
**Pull request by chenjiahan**
### <samp>🤖 Generated by Copilot at 4acf59f</samp>

This pull request updates the license files of the project and its packages to reflect the current year and status, and creates a separate file for the third-party licenses. This improves the clarity and simplicity of the licensing information, and acknowledges the external libraries that the project depends on.

### <samp>🤖 Generated by Copilot at 4acf59f</samp>

*  Update the year in the LICENSE files of all packages to indicate project maintenance ([link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-183f41c28d9e1eae637fbf439f9b6a2c0da95f9008272be0277ca2197e59568fL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-41d5ee2732d84e131c9dd6aa5bda6a383355ef81c34a795c9e55ada8be5686edL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-8879974de1f75a89be1999087dd7de6416eca7202ce6bab136db7b30617682b7L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-92fd6428324d2a1a1ebb7ed690a93378648005647057349e96c1e21d71637283L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-5e4e6115a1c4e547142839a4089f9332b67dc915ffcb476e18af684a2fcd45bdL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-77f19df731040ec58a04b6450402f9da242a86b9954c517e0adf573c3952f552L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-a8d8d33e777e3e1065817b8b0abafa7e28d521998f7613e04c8f35356babb211L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-ffe6f067e8adcab1ecad8eed225bceabc206b6f41bfe082e0d9e4667fa52c8f1L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-f49bb8c453dbc4d529cb38e15c6a3f27a7097297e40a71d1b7823eb166758f1aL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-a322d7e444ce17adad99a26dc931c41afc5520fa214d4aeb7c79c85ff1f096d9L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-17d6c789029b63be0377bcfa56dc19876e4448b36933c855adbf62cded1264ebL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-4563ef58cc80e8c2b4245b7a850510bbaf55e5ace8a78f871f40c4a27e16ff3aL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-d255a5ff579ee3beb9a36120cbaf373ded083b1012a82a337f9726f7702a4246L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-dcf0ed7c14618658db26a1e440b5b937f95f27a8dd0d208c49a9ac9e6c33b0feL3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-c24e1a7156e01b3288d1ab0b7d16d2c85848b6ff9f436e800ce08475314c7d32L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-cea709afd0b2cc690168e99542c15c2cf5268a4e5457fed632aab9268e33e898L3-R3))
*  Move the third-party licenses from the main `LICENSE` file to a separate `THIRD-PARTY-LICENSE` file for clarity and conciseness ([link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7L22-L232), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-cc2f2fe6b450c228fbed321e6df533fbce3b810b1e506f93cd509fc43fe4594bR1-R206))
*  Add a reference to the `THIRD-PARTY-LICENSE` file in the `README.md` and `README.zh-CN.md` files under the License section to inform users and contributors ([link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R81-R82), [link](https://github.com/web-infra-dev/modern.js/pull/4488/files?diff=unified&w=0#diff-13f564fa6ce92394f8423bbcea359aef7e9ebc4f79763efc0d791d9ce5b6f987R80-R81))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-shared/LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2021 Modern.js
+Copyright (c) 2021-present Modern.js
```
[the remark this answers is not in the document]
Comment (chenjiahan) [reply packages/builder/builder-shared/LICENSE:3]: 当前时间哈

**Review by Asuka109 — APPROVED** [review]

File: packages/builder/builder-shared/LICENSE (hunk 1 above)
[the remark this answers is not in the document]
Comment (Asuka109) [reply packages/builder/builder-shared/LICENSE:3]: 这样标注在其它项目里有例子嘛

[the remark this answers is not in the document]
Comment (chenjiahan) [reply packages/builder/builder-shared/LICENSE:3]: 很常见吧
Vue: https://github.com/vuejs/core/blob/main/LICENSE
Nuxt: https://github.com/nuxt/nuxt/blob/main/LICENSE
Rspack: https://github.com/web-infra-dev/rspack/blob/main/LICENSE

[the remark this answers is not in the document]
Comment (Asuka109) [reply packages/builder/builder-shared/LICENSE:3]: get

**Pull request merged**